services/security/authentication/casigned/casigned.go

```diff
@@ -37,6 +44,47 @@ func (c *client) Login(ctx context.Context, group string, identity *security.Ide
 	return &response.Token, nil
 }
 
+func RenewRoutine(group, server string) {
+	renewalAttempt := 0
+	// Waiting for a few seconds to avoid spamming short-lived sdk user
+	time.Sleep(time.Second * 5)
+	for {
+		wssdConfig := auth.WssdConfig{}
+		err := marshal.FromJSONFile(auth.GetWssdConfigLocation(), &wssdConfig)
+		if err != nil {
+			fmt.Printf("Failed to open config file in location %s: %v\n", auth.GetWssdConfigLocation(), err)
+			panic("Failed to open config file")
+		}
+
+		sleepTime, err := renewTime(wssdConfig.ClientCertificate)
+		if err != nil {
+			fmt.Printf("Failed to find sleep time for cert %v \n", err)
+			panic("Failed to find renew time for certificate")
+		}
+		fmt.Printf("Waiting for %v to renew cert\n", sleepTime)
+		time.Sleep(sleepTime)
+		fmt.Printf("Attempting to renew certificate\n")
+		err = auth.RenewCertificates(server, auth.GetWssdConfigLocation())
+		if err != nil {
+			if renewalAttempt > 5 {
+				fmt.Printf("Failed to renew cert to %v \n", err)
+				panic("Failed to renew certificate")
```

Comment:
Once we fix the sleep timer, we dont have to panic here.
Panic should be done only if the cert expired, and we want to force app to login again to get a new certificate.
Remember, in this context the cert is still valid and can communicate with MOC